chakra-wrapper/JsEngine.h

```diff
@@ -525,26 +511,24 @@ class JsValue
   class JsFunctionArgumentsImpl : public JsFunctionArguments
   {
   public:
-    JsFunctionArgumentsImpl(JsValueRef* arr_, size_t n_,
-                            const JsValue& missingArgValue_)
+    JsFunctionArgumentsImpl(JsValueRef* arr_, size_t n_)
       : arr(arr_)
       , n(n_)
-      , missingArgValue(missingArgValue_)
     {
+      undefined = std::make_unique<JsValue>(JsValue::Undefined());
     }
 
-    size_t GetSize() const noexcept { return n; }
+    size_t GetSize() const noexcept override { return n; }
 
-    const JsValue& operator[](size_t i) const noexcept
+    const JsValue& operator[](size_t i) const noexcept override
     {
-      return i < n ? reinterpret_cast<const JsValue&>(arr[i])
-                   : missingArgValue;
+      return i < n ? reinterpret_cast<const JsValue&>(arr[i]) : *undefined;
```

Comment:
`reinterpret_cast` is probably ok and does right job now, but I guess it can break in an unexpected way in the future. Maybe replace with constructor?

Reply:
Added a comment